leetcode/linked_list_cycle/step3.py

```diff
@@ -0,0 +1,20 @@
+from typing import Optional
+
+# Definition for singly-linked list.
+class ListNode:
+    def __init__(self, x):
+        self.val = x
+        self.next = None
+
+class Solution:
+    def hasCycle(self, head: Optional[ListNode]) -> bool:
+        visited_node_set = set()
```

Comment:
```suggestion
        visited_node = set()
```

変数名に`set`のように型の情報を入れるのはハンガリアン記法といってアンチパターンの一つだそうです。
個人的には`visited_node`で良いと思いました。
https://ja.wikipedia.org/wiki/%E3%83%8F%E3%83%B3%E3%82%AC%E3%83%AA%E3%82%A2%E3%83%B3%E8%A8%98%E6%B3%95

Reply:
指摘していただいて、型を変数名に入れるような記法に慣れてしまっているなと気づきました。ありがとうございます！  

要素の存在確認を行うので複数形の`visited_nodes`として修正してみましたが、どうでしょうか